codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java

```diff
@@ -374,10 +372,18 @@ public String stringShape(final StringShape shape) {
         shape,
         context.symbolProvider().toSymbol(shape)
       );
-
+      var noEnumMatchedCheck = "";
       if (this.isOptional) {
         someWrapIfRequired = "Wrappers.Companion_Option_.Create_Some_(%s)";
         returnType = "Wrappers.Option";
+        // In AWS SDK, some shapes don't have required trait and also don't have pointers in it.
+        // This will result the default value of the string be "" if not provided.
+        noEnumMatchedCheck =
```

Comment:
Let's be defensive and error out in case of it being required and not matching an enum value.